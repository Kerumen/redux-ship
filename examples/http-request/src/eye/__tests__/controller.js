// @flow
import 'babel-polyfill';
import * as Ship from 'redux-ship';
import * as EyeController from '../controller';

const config = [
  {"action":{"type":"Load"},"snapshot":[{"type":"Commit","commit":{"type":"LoadStart"}},{"type":"Effect","effect":{"type":"HttpRequest","url":"http://swapi.co/api/people/3/"},"result":"{\"name\":\"R2-D2\",\"height\":\"96\",\"mass\":\"32\",\"hair_color\":\"n/a\",\"skin_color\":\"white, blue\",\"eye_color\":\"red\",\"birth_year\":\"33BBY\",\"gender\":\"n/a\",\"homeworld\":\"http://swapi.co/api/planets/8/\",\"films\":[\"http://swapi.co/api/films/5/\",\"http://swapi.co/api/films/4/\",\"http://swapi.co/api/films/6/\",\"http://swapi.co/api/films/3/\",\"http://swapi.co/api/films/2/\",\"http://swapi.co/api/films/1/\",\"http://swapi.co/api/films/7/\"],\"species\":[\"http://swapi.co/api/species/2/\"],\"vehicles\":[],\"starships\":[],\"created\":\"2014-12-10T15:11:50.376000Z\",\"edited\":\"2014-12-20T21:17:50.311000Z\",\"url\":\"http://swapi.co/api/people/3/\"}"},{"type":"Commit","commit":{"type":"LoadSuccess","color":"red"}}]}
];

test('controller', () => {
  config.forEach(({action, snapshot}) => {
    expect(Ship.simulate(EyeController.control(action), snapshot)).toEqual(snapshot);
  });
});
