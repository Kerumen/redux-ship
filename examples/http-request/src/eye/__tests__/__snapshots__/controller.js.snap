exports[`test controller with generated snapshot 1`] = `
Object {
  "result": undefined,
  "snapshot": Array [
    Object {
      "commit": Object {
        "type": "LoadStart",
      },
      "type": "Commit",
    },
    Object {
      "effect": Object {
        "type": "HttpRequest",
        "url": "http://swapi.co/api/people/3/",
      },
      "result": "{\"eye_color\":\"red\"}",
      "type": "Effect",
    },
    Object {
      "commit": Object {
        "color": "red",
        "type": "LoadSuccess",
      },
      "type": "Commit",
    },
  ],
}
`;
